v2/lints/mozilla/lint_mp_pss_parameters_encoding_correct_test.go

```diff
@@ -0,0 +1,70 @@
+package mozilla
+
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"testing"
+
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/test"
+)
+
+func TestPssAidEncoding(t *testing.T) {
+	testCases := []struct {
+		Name           string
+		InputFilename  string
+		ExpectedResult lint.LintStatus
+	}{
+		{
+			Name:           "Standard RSASSA-PSS with SHA256",
+			InputFilename:  "rsassapssWithSHA256.pem",
+			ExpectedResult: lint.Pass,
+		},
+		{
+			Name:           "Standard RSASSA-PSS with SHA256 but the hash parameters are empty instead of NULL",
+			InputFilename:  "rsassapssWithSHA256EmptyHashParams.pem",
+			ExpectedResult: lint.Error,
+		},
+		{
+			Name:           "Standard RSASSA-PSS with SHA384",
+			InputFilename:  "rsassapssWithSHA384.pem",
+			ExpectedResult: lint.Pass,
+		},
+		{
+			Name:           "Standard RSASSA-PSS with SHA384 but the hash parameters are empty instead of NULL",
+			InputFilename:  "rsassapssWithSHA384EmptyHashParams.pem",
+			ExpectedResult: lint.Error,
+		},
+		{
+			Name:           "Standard RSASSA-PSS with SHA512",
+			InputFilename:  "rsassapssWithSHA512.pem",
+			ExpectedResult: lint.Pass,
+		},
+		{
+			Name:           "Standard RSASSA-PSS with SHA512 but the hash parameters are empty instead of NULL",
```

Comment:
Suggestion: These variations all only exercise a single mis-encoding (empty hash parameters). Would it be useful to include explicit hash parameters that are more esoteric, such as a different mask length?

I'm not sure if I'm underestimating the difficulty, so I thought I'd check here.

Reply:
Added a test case for irregular (maybe invalid or unsupported is a better name for this?) salt length. Other test cases (e.g. irregular trailerField) are indeed more work to implement also because the crypto libraries do not support it out-of-the-box.

Reply:
@mtgag Thanks for adding another test case. I think the coverage in-branch is sufficient to see the lint merged. It would be nice to have more malformed test cases but it sounds like the difficulty would mean holding up the PR longer and I'd prefer to see it merged and iterated on as time allows.

Since @sleevi gave this branch a +1 without requesting the test case change as blocking feedback I'm going to go ahead and merge. @mtgag If you think you'd be up to adding the more difficult test cases would you mind filing an issue to act as a marker/discussion point?

Thanks all!